tests/user_input_checks_test.py

```diff
@@ -435,3 +436,43 @@ def test_passing_custom_density_estimator(arg):
         density_estimator = arg
     prior = MultivariateNormal(torch.zeros(2), torch.eye(2))
     _ = SNPE_C(prior=prior, density_estimator=density_estimator)
+
+
+def test_validate_theta_and_x_cpu():
+
+    cpu_device = torch.device('cpu')
+
+    theta = torch.ones((32,8), dtype=torch.float32).to(cpu_device)
+    x = torch.zeros((32,100), dtype=torch.float32).to(cpu_device)
+    otheta = torch.FloatTensor((32,8)) #using an explicit type
+
+    #check assumptions in implementation
```

Comment:
replace this line comment by short explanations about the fail after the asserts below?

Reply:
fixed with 3c11ccd